diff --git a/configure b/configure
index f017ca255..c0c7cbd17 100755
--- a/configure
+++ b/configure
@@ -123,20 +123,18 @@ echo "checking SCons version ... $scons_v"
 echo "Running RSFROOT=$root $scons $args config ..."
 rm -rf env.*sh .scon* config.py
 
-first=1
-arglist=""
-for i in "$@"
-do
-    if test $first -ne $prefixparnum; then
-        arglist=""$arglist" \"\$"$first"\""
+args=()
+i=1
+for arg in "$@"; do
+    if [ "$i" -ne "$prefixparnum" ]; then
+        args+=("$arg")   # preserves spaces exactly
     fi
-    first=`expr $first + 1` 
+    ((i++))
 done
+args+=(config)
 
 echo "------------------------"
-command1=`printf "$arglist config"`
-command2="RSFROOT=$root $scons `printf "$command1"`"
-eval $command2
+RSFROOT="$root" "$scons" "${args[@]}"
 echo "------------------------"
 echo "Done with configuration."
 
